catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/cache/solr/impl/CachingFederationStrategy.java

```diff
@@ -546,14 +545,20 @@ private QueryRequest getQueryRequest() {
     }
 
     private SourceResponse cloneResponse(SourceResponse sourceResponse) {
-      List<Result> clonedResults =
-          sourceResponse
-              .getResults()
-              .stream()
-              .map(Result::getMetacard)
-              .map(m -> new MetacardImpl(m, m.getMetacardType()))
-              .map(ResultImpl::new)
-              .collect(Collectors.toList());
+      List<Result> clonedResults = new ArrayList<>();
+      sourceResponse
+          .getResults()
+          .forEach(
+              result -> {
```

Comment:
✏️ Large lambdas should be extracted as methods. If you do so, you'll have the ability to set the return type of the method to a `Result` so you'd be able to go back to a `map` then `collect` stream structure.